io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java

```diff
@@ -70,6 +71,33 @@ public Gui(BukkitEventNode bukkitEventNode, Component title) {
         this.maxSlots = -1;
     }
 
+    @SuppressWarnings("deprecation")
+    public void updateTitle(@NotNull Player player, @NotNull Component title) {
+        this.title = title;
+
+        if (!this.isOpening() || this.titleUpdating)
+            return;
+
+        this.titleUpdating = true;
+
+        MCPlayer mcPlayer = MCPlayer.from(player);
+        player.closeInventory();
+
+        MCSchedulers.getEntityScheduler(player).schedule(() -> {
```

Comment:
Running a scheduler is way overkill here, by default opening a new inventory closes the previous one.

You can get away with something like
```java
    public void updateTitle(@NotNull Player player, @NotNull Component title) {
        this.title = title;

        if (!this.isOpening() || this.titleUpdating)
            return;

        this.titleUpdating = true;

        MCPlayer mcPlayer = MCPlayer.from(player);
        try {
            this.inventory = Bukkit.createInventory(null, this.getRows() * 9, MCAdventure.asLegacyString(this.title, mcPlayer.getLocale()));
            this.renderSlots(player);
            player.openInventory(this.inventory);
        } finally {
            this.titleUpdating = false;
        }
    }
```